homeassistant/components/yeelight/light.py

```diff
@@ -707,23 +707,33 @@ def turn_on(self, **kwargs) -> None:
         flash = kwargs.get(ATTR_FLASH)
         effect = kwargs.get(ATTR_EFFECT)
 
+        turn_on_before_change = True
+
         duration = int(self.config[CONF_TRANSITION])  # in ms
         if ATTR_TRANSITION in kwargs:  # passed kwarg overrides config
             duration = int(kwargs.get(ATTR_TRANSITION) * 1000)  # kwarg in s
 
-        self.device.turn_on(
-            duration=duration,
-            light_type=self.light_type,
-            power_mode=self._turn_on_power_mode,
-        )
-
-        if self.config[CONF_MODE_MUSIC] and not self._bulb.music_mode:
-            try:
-                self.set_music_mode(self.config[CONF_MODE_MUSIC])
-            except BulbException as ex:
-                _LOGGER.error(
-                    "Unable to turn on music mode, consider disabling it: %s", ex
-                )
+        # When in music mode, skip turn_on if there is no change, except brightness, to
+        # speed up color changes.
+        # changes
+        if self._bulb.music_mode:
+            if colortemp or rgb or flash or effect:
+                turn_on_before_change = False
+        else:
```

Comment:
Is it harmful to turn on music mode if already in music mode? Usually we try to avoid those kinds of checks since they make it impossible to change state if we'd have incorrect state info.

Reply:
Yeelight library doesn't allow to turn on music_mode again: https://gitlab.com/stavros/python-yeelight/-/blob/master/yeelight/main.py#L806 . Probably becaise of that socket handling. Music mode reverses connection, so bulb connects to HA.

Reply:
Ok.